compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/CombinedComposeScene.kt

```diff
@@ -0,0 +1,623 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.scene
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Composition
+import androidx.compose.runtime.CompositionContext
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.InternalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.focus.FocusDirection
+import androidx.compose.ui.focus.FocusManager
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.graphics.BlendMode
+import androidx.compose.ui.graphics.Canvas
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.key.KeyEvent
+import androidx.compose.ui.input.key.KeyInputElement
+import androidx.compose.ui.input.pointer.PointerButton
+import androidx.compose.ui.input.pointer.PointerEventType
+import androidx.compose.ui.input.pointer.PointerInputEvent
+import androidx.compose.ui.input.pointer.PointerType
+import androidx.compose.ui.node.LayoutNode
+import androidx.compose.ui.node.RootNodeOwner
+import androidx.compose.ui.platform.PlatformContext
+import androidx.compose.ui.platform.setContent
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntRect
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.toIntRect
+import androidx.compose.ui.util.fastAny
+import androidx.compose.ui.util.fastForEach
+import androidx.compose.ui.util.fastForEachReversed
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.Dispatchers
+
+/**
+ * Constructs a combined [ComposeScene] using the specified parameters. Unlike the regular
+ * [ComposeScene], this version doesn't employ [ComposeSceneContext.createPlatformLayer] to
```

Comment:
Cmd-clicking `[ComposeScene]` here doesn't actually take you to the "regular" compose scene, which is confusing. Maybe the constructor function for the "regular" compose scene should have a more meaningful name after all.

Reply:
> Maybe the constructor function for the "regular" compose scene should have a more meaningful name after all.

I am also lean towards this approach. Until we remove `CombinedComposeScene` completely, we should have a distinguishable prefix.

Reply:
On the other hand, if we are 95% sure that `CombinedComposeScene` is only for compatibility and we'll remove it in the future, then `ComposeScene` costructor makes sense.

But as the ComposeScene API can be changed easely now, I suggest to name functions taking into account the current code, not the future code.